File: Recursion/recursion.cpp
// recursive factoring function in C++

#include <iostream>
using namespace std;

int factor(int num, int div) //intake number and divisor (factor)
{
  int fac = 0;
  if((num % div) == 0) {
    fac = factor(num/div, div);
    fac++;
  }
  return(fac);
}

int main()
{
  int number, divisor, fac_count;
  cout << "Enter number to be factored: ";
  cin >> number;
  cout << "Enter factor: ";
  cin >> divisor;

  fac_count = factor(number,divisor);
  cout << divisor << " factors into " << number << " " << fac_count << " times." << endl;

  return(0);
}

File: Recursion/Recursion.java
// The object of this program is to perform a recursive factoring operation to determine the degree of a given factor of an integer.
// The Scanner class does the legwork of managing user input.
import java.util.Scanner;

// Important that your .java file have a public class matching the filename.
public class Recursion {
  static int factor(int num, int div) {
    int fac = 0;
    if(num % div == 0) {
      fac = factor(num/div, div);
      fac++;
    }
    return fac;
  }

  public static void main(String[] args) {
    Scanner myScanner = new Scanner(System.in);

    System.out.print("Enter number to be factored: ");
    int number = myScanner.nextInt();
    System.out.print("Enter factor: ");
    int divisor = myScanner.nextInt();

    System.out.println(divisor + " factors into " + number + " " + factor(number, divisor) + " times.");
  }
}
